core/app/models/spree/price.rb

```diff
@@ -12,7 +13,12 @@ class Price < Spree::Base
       less_than_or_equal_to: MAXIMUM_AMOUNT
     }
 
-    after_save :set_default_price
+    scope :valid_before, -> (date) { where("#{self.table_name}.valid_from <= ?", date).order(valid_from: :desc) }
```

Comment:
I like to write this as `where("#{Spree::Price.table_name}.valid_from <= ?", date)` nowadays.